backend/storage-service/src/jvmMain/kotlin/file/services/SynchronizationService.kt

```diff
@@ -0,0 +1,308 @@
+                val internalFile = File(fsPath, folder.path)
+                if (!internalFile.exists() || !internalFile.isDirectory) {
+                    throw RPCException.fromStatusCode(HttpStatusCode.NotFound)
+                }
+
+                val syncType = if (aclService.hasPermission(folder.path, actor.username, AccessRight.WRITE)) {
+                    SynchronizationType.SEND_RECEIVE
+                } else if (aclService.hasPermission(folder.path, actor.username, AccessRight.READ)) {
+                    SynchronizationType.SEND_ONLY
+                } else {
+                    throw RPCException.fromStatusCode(HttpStatusCode.Unauthorized)
+                }
+
+                if (CephFsFastDirectoryStats.getRecursiveFileCount(internalFile) > 1_000_000) {
+                    throw RPCException(
+                        "Number of files in directory exceeded for synchronization",
+                        HttpStatusCode.Forbidden
+                    )
+                }
+
+                val id = UUID.randomUUID().toString()
+                val device = chooseFolderDevice(session)
+                affectedDevices.add(device)
+
+                Mounts.mount.call(
+                    MountRequest(
+                        listOf(MountFolder(id, folder.path))
+                    ),
+                    authenticatedClient
+                ).orRethrowAs {
+                    throw RPCException.fromStatusCode(HttpStatusCode.InternalServerError, "Failed to prepare folder for synchronization")
+                }
+
+                ids.add(id)
+                devices.add(device)
+                syncTypes.add(syncType)
+            }
+
+            session.sendPreparedStatement(
+                {
+                    setParameter("ids", ids)
+                    setParameter("devices", devices.map { it.id })
+                    setParameter("paths", request.items.map { folder -> folder.path })
+                    setParameter("users", ids.map { actor.username })
+                    setParameter("access", syncTypes.map { it.name })
+                },
+                """
+                    insert into storage.synchronized_folders(
+                        id, 
+                        device_id, 
+                        path,
+                        user_id,
+                        access_type
+                    ) values (
+                        unnest(:ids::text[]),
+                        unnest(:devices::text[]),
+                        unnest(:paths::text[]),
+                        unnest(:users::text[]),
+                        unnest(:access::text[])
+                    )
+            """
+            ).rowsAffected
+        }
+
+        affectedDevices.forEach { device ->
+            val mounter = Mounts.ready.call(Unit, authenticatedClient)
+            if (mounter.statusCode != HttpStatusCode.OK || !mounter.orThrow().ready || !syncthing.isReady(device)) {
```

Comment:
I would probably add a comment here to explain the difference between the mounter being ready and syncthing being ready.